aten/src/ATen/core/List.h

```diff
@@ -53,6 +53,28 @@ bool operator==(const ListElementReference<T, Iterator>& lhs, const T& rhs);
 template<class T, class Iterator>
 bool operator==(const T& lhs, const ListElementReference<T, Iterator>& rhs);
 
+template<class T>
+struct ListElementConstReferenceTraits {
+  // In the general case, we cannot expose a true const reference to
+  // the contents of an IValue, so we copy.
+  using const_reference = T;
+};
+
+template<>
+struct ListElementConstReferenceTraits<std::string> {
+  using const_reference = const std::string&;
+};
+
+template<>
```

Comment:
We could take this a bit further for all types with `std::is_reference_v<ListElementConstReferenceTraits<T>::const_reference>`, for example `optional<Tensor>` could benefit from this.